requirements.txt

```diff
@@ -1,5 +1,5 @@
 netmiko==1.4.3
-git+https://github.com/StackStorm/PySwitchLib.git@v1.2.1#egg=pyswitchlib
-requests==2.18.4
+-e git+https://github.com/StackStorm/PySwitchLib.git@bb268e3c2f9b0eee690d939124b5c260993ebe57#egg=pyswitchlib
```

Comment:
How come this is a commit and not a version number?

Reply:
Also, which branch is this commit located?  I don't see this on master branch of pyswitchlib.

Reply:
This is branched from v1.2.1 in PySwitchLib. This network_essentials pack pointed to v1.2.1 and not master. Hence, since I am not aware of why it was not pointing to master originally and no way for me to test the pack without the devices and all, it was safest to keep it the same and make changes off of v1.2.1 branch. I created a new branch https://github.com/StackStorm/PySwitchLib/tree/upgradeRequestsLibVersion off of v1.2.1. 

The reason it is pointing to a commit and not the version is because I am not sure if we should up the version already with every commit. Hence, pointing it to a commit for now, and later when there are enough changes in PySwitchLib which warrants upping the version, we can change to the new version in the requirements of network_essentials. Basically, trying to follow how versions are upped in orquesta.

Reply:
We do an official release for Orquesta when releasing a new st2 version.  We only use the commit when we're still in development.

In this case for pySwitchLib, you can create a branch off of v1.2.1 and name it v1.2.2 with your changes. Then you can reference v1.2.2 here.  As it stands in the PR right now, no body will know where this commit comes from, especially under an obscure branch named "upgradeRequestsLibVersion".

Reply:
You should still create a PR in PySwitchLib and merge the change into master.  Because whoever is going to pick up PySwitchLib will pick it up from master and will not have your change when they create a new version.

Reply:
My plan was to merge the upgradeRequestsLibVersion in to v1.2.1 (PySwitchLib) itself and keep the requirements (network_essentials) to still point to v1.2.1 after the merge.

I had to point to the new commit in network_essentials for now as upping the requests lib version in PySwitchLib is a must for successfully installing network_essentials pack with the latest requests lib version. 

Do you agree with this approach? or you want to up the version to 1.2.2 for PySwitchLib?

Reply:
> I had to point to the new commit in network_essentials for now as upping the requests lib version in PySwitchLib is a must for successfully installing network_essentials pack with the latest requests lib version.

I don't agree with this approach.  This is working backward.  If network_essentials depend on PySwitchLib, then fix PySwitchLib and get it in order first.

Also, since you making a change and we don't know what is using v1.2.1 of PySwitchLib, I would not recommend merging the upgradeRequestsLibVersion branch into v1.2.1.  Like I said, create a v1.2.2 branch and merge the change there and also to master branch.

Finally, then update network_essentials to point to v1.2.2 branch of PySwitchLib.

Reply:
@m4dcoder I have updated the requirements.txt to point to v1.2.2 branch of PySwitchLib now.